pycode/memilio-epidata/memilio/epidata/getPopulationData.py

```diff
@@ -240,286 +237,71 @@ def get_population_data(read_data=dd.defaultDict['read_data'],
-        key = np.zeros(len(zensus))
-        for i in range(len(key)):
-            for j in range(len(reg_key)):
-                if zensus['Name'].values[i] == reg_key['NAME'].values.astype(
-                        str)[j]:
-                    if zensus[dd.invert_dict(dd.GerEng)[dd.EngEng['population']]].values[i] == round(
-                            reg_key['Zensus_EWZ'].values[j] * 1000):
-                        key[i] = reg_key['AGS'].values[j]
-
-        inds = np.unique(key, return_index=True)[1]
-        # columns of downloaded data which should be replaced
-        male = ['M_Unter_3', 'M_3_bis_5', 'M_6_bis_14', 'M_15_bis_17',
-                'M_18_bis_24', 'M_25_bis_29', 'M_30_bis_39', 'M_40_bis_49',
-                'M_50_bis_64', 'M_65_bis_74', 'M_75_und_aelter']
-        female = ['W_Unter_3', 'W_3_bis_5', 'W_6_bis_14', 'W_15_bis_17',
-                  'W_18_bis_24', 'W_25_bis_29', 'W_30_bis_39', 'W_40_bis_49',
-                  'W_50_bis_64', 'W_65_bis_74', 'W_75_und_aelter']
-
-        if not split_gender:
-            # get data from zensus file and add male and female population data
-            data = np.zeros((len(inds), len(male)+2))
-            data[:, 0] = key[inds].astype(int)
-            data[:, 1] = zensus[dd.invert_dict(
-                dd.GerEng)[dd.EngEng['population']]].values[inds].astype(int)
-            for i in range(len(male)):
-                data[:, i+2] = zensus[male[i]].values[inds].astype(
-                    int) + zensus[female[i]].values[inds].astype(int)
-
-            # define new columns for dataframe
-            columns = [
-                dd.EngEng['idCounty'],
-                dd.EngEng['population'],
-                '<3 years', '3-5 years', '6-14 years', '15-17 years',
-                '18-24 years', '25-29 years', '30-39 years', '40-49 years',
-                '50-64 years', '65-74 years', '>74 years']
+    directory = os.path.join(out_folder, 'Germany')
+    gd.check_dir(directory)
+
+    df_pop_raw = read_population_data(username, password, read_data, directory)
+
+    column_names = list(df_pop_raw.columns)
+    # rename columns
+    rename_columns = {
+        column_names[0]: dd.EngEng['idCounty'],
+        column_names[1]: dd.EngEng['county'],
+        column_names[2]: dd.EngEng['ageRKI'],
+        column_names[3]: dd.EngEng['number'],
+        column_names[4]: dd.EngEng['male'],
+        column_names[5]: dd.EngEng['female']
+    }
+    df_pop_raw.rename(columns=rename_columns, inplace=True)
+    # remove date and explanation rows at end of table
+    df_pop_raw = df_pop_raw[:np.where(df_pop_raw[dd.EngEng['idCounty']].str.contains(
+        '__') == True)[0][0]].reset_index(drop=True)
+    # get indices of counties first lines
+    idCounty_idx = []
+    for id in df_pop_raw[dd.EngEng['idCounty']].dropna().unique():
+        idCounty_idx.append(
+            df_pop_raw[df_pop_raw[dd.EngEng['idCounty']] == id].index.min())
```

Comment:
```suggestion
    idCounty_idx = df_pop_raw.groupby(dd.EngEng['idCounty']).head(1).index.tolist()
```
When possible, try not to use loops :)